starport/pkg/spn/spn.go

```diff
@@ -274,6 +275,34 @@ func (c *Client) broadcast(ctx context.Context, clientCtx client.Context, msgs .
 
 	c.out.Reset()
 
+	// calculate the necessary gas for the transaction
+	txf, err := tx.PrepareFactory(clientCtx, c.factory)
+	if err != nil {
+		return err
+	}
+	c.factory = txf
```

Comment:
Is modifying factory in `Client` required here? Because it may introduce some side effect for other callers of `Client`.

Reply:
Yes without `PrepareFactory ` `CalculateGas` fails

From the doc:
```
// PrepareFactory ensures the account defined by ctx.GetFromAddress() exists and
// if the account number and/or the account sequence number are zero (not set),
// they will be queried for and set on the provided Factory. A new Factory with
// the updated fields will be returned.
```
I don't think it has any bad side effect

Reply:
Should avoid modifying `c.factory` anyway and only use returned new factory for the underlying operation?